src/main/java/com/sid/gl/services/impl/UserService.java

```
@@ -156,19 +174,63 @@ public String verifyCode(FactorRequest request) throws BadRequestException {
                           new InternalServerException("unable to generate access token"));
      }
 
+     //pour une nouvelle localisation en ajoutant le token selon la localisation
      @Override
      public NewLocationToken isNewLocation(String username, String ip) {
+          if(!isGeoIpLibEnabled()) {
+               return null;
+          }
+          try {
+               final InetAddress ipAddress = InetAddress.getByName(ip);
+               final String country = databaseReader.country(ipAddress)
+                       .getCountry()
+                       .getName();
+               System.out.println(country + "====****");
+               final User user = userRepository.findUserByUsername(username).orElseThrow(()->new UserNotFoundException("User by username not found!!!"));
+               final UserLocation loc = userLocationRepository.findByUserAndCountry(user,country);
+               if ((loc == null) || !loc.isEnabled()) {
+                    return createNewLocationToken(country, user);
+               }
+          } catch (final Exception e) {
```

add Logger here for exception message. Not swallow Exception here